bundle/edu.gemini.catalog/src/main/scala/edu/gemini/catalog/image/ImageCatalogClient.scala

```diff
@@ -0,0 +1,266 @@
+  def halfRa = (ra / 2).getOrElse(Angle.zero)
+}
+
+case object AngularSize {
+  val zero = AngularSize(Angle.zero, Angle.zero)
+
+  /** @group Typeclass Instances */
+  implicit val equal = Equal.equalA[AngularSize]
+
+  /** @group Typeclass Instances */
+  implicit val order: Order[AngularSize] =
+    Order.orderBy(a => (a.ra.toRadians * a.dec.toRadians))
+}
+
+/**
+  * Query to request an image for a catalog and coordinates
+  */
+case class ImageSearchQuery(catalog: ImageCatalog, coordinates: Coordinates, size: AngularSize) {
+  import ImageSearchQuery._
+
+  def url: NonEmptyList[URL] = catalog.queryUrl(coordinates)
+
+  def fileName(extension: String): String = s"img_${catalog.id}_${coordinates.toFilePart}_${size.toFilePart}.$extension"
+
+  def isNearby(query: ImageSearchQuery): Boolean =
+    catalog === query.catalog && isNearby(query.coordinates)
+
+  def isNearby(c: Coordinates): Boolean =
+    coordinates.angularDistance(c) <= maxDistance
+}
+
+object ImageSearchQuery {
+  /** @group Typeclass Instances */
+  implicit val equals: Equal[ImageSearchQuery] = Equal.equalA[ImageSearchQuery]
+
+  /** Harcoded value of the max FoV for any Gemini instrument */
+  val maxDistance: Angle = Angle.fromArcmin(4.5)
+
+  implicit class DeclinationShow(val d: Declination) extends AnyVal {
+    def toFilePart: String = Declination.formatDMS(d, ":", 2)
+  }
+
+  implicit class RightAscensionShow(val a: RightAscension) extends AnyVal {
+    def toFilePart: String = a.toAngle.formatHMS
+  }
+
+  implicit class CoordinatesShow(val c: Coordinates) extends AnyVal {
+    def toFilePart: String = s"ra_${c.ra.toFilePart}_dec_${c.dec.toFilePart}"
+  }
+
+  implicit class AngularSizeShow(val s: AngularSize) extends AnyVal {
+    def toFilePart: String = f"w_${s.ra.toArcsecs.toInt}_h_${s.dec.toArcsecs.toInt}"
+  }
+}
+
+/**
+  * Image in the file system
+  */
+case class ImageInFile(query: ImageSearchQuery, file: Path, fileSize: Long) {
+
+  /**
+    * Tests if the image contains the coordinates parameter
+    */
+  def contains(c: Coordinates): Boolean = {
+    val ε = query.catalog.adjacentOverlap
+    // Convert everything to radians, calculations in Angle-space don't work due to range overflow
+    // Image size
+    val εφ = (query.size.halfDec.toRadians - ε.toRadians).max(0)
+    val ελ = (query.size.halfRa.toRadians  - ε.toRadians).max(0)
+
+    // target coordinates
+    val φ = toRadians(c.dec.toDegrees)
+    val λ = c.ra.toAngle.toRadians
+
+    // image coordinates
+    val φ0 = toRadians(query.coordinates.dec.toDegrees)
+    val λ0 = query.coordinates.ra.toAngle.toRadians
+
+    val θ = cos(φ0)
+
+    // Distance
+    val Δφ = abs(φ - φ0)
+    val Δλ = abs(λ - λ0)
+
+    Δφ <= εφ && Δλ*θ <= ελ
+  }
+}
+
+object ImageInFile {
+  /** @group Typeclass Instances */
+  implicit val equals: Equal[ImageInFile] = Equal.equalA[ImageInFile]
+
+  val fileRegex: Regex = """img_(.*)_ra_(.*)_dec_(.*)_w_(\d*)_h_(\d*)\.fits.*""".r
```

Comment:
The pattern trusts that only files written by the program will match, you can probably fool it if you want